install-script.sh

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env bash
+
+# This file is part of MinIO DirectPV
+# Copyright (c) 2022 MinIO, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+echo "Installing direct PV"
+
+operating_system=""
+darwin="Darwin"
+linux="Linux"
+windows="Windows"
+
+
+function directpv(){
+    operating_system=$(fetch_OS)
+    verify_kubectl $operating_system
+    verify_krew $operating_system
+    kubectl krew install directpv
+    kubectl directpv install
+}
+
+# Fetch the operating system
+function fetch_OS(){
+    if [ "$(uname)" == "Darwin" ]; then
+        operating_system=$darwin
+    elif [ "$(expr substr $(uname -s) 1 5)" == "Linux" ]; then
+        operating_system=$linux
+    elif [ "$(expr substr $(uname -s) 1 10)" == "MINGW32_NT" ]; then
+        operating_system=$windows
+    elif [ "$(expr substr $(uname -s) 1 10)" == "MINGW64_NT" ]; then
+        operating_system=$windows
+    fi
+    echo "$operating_system"
+}
+
+
+function verify_kubectl() {
+    # check for kubectl installation
+    kubectl version --client
+    # Install kubectl if not present
+    if [ $? -ne 0 ]; then
+        install_kubectl "$1"
+    fi
+    # Check for kubectl setup
+    kubectl version
+    # Install minikube and configure kubectl to use it
+    if [ $? -ne 0 ]; then
+        verify_minikube "$1"
+    fi
+}
+
+function install_kubectl() {
+    if [ "$1" == $darwin ]; then
+        cd /usr/local/bin/
+        curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/darwin/amd64/kubectl"
+        chmod +x kubectl
+        cd -
+    elif [ "$1" == $linux ]; then
+        cd ~/.local bin
+        curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+        chmod +x kubectl
+        cd -
+    elif [ "$1" == $windows ]; then
+        curl -LO "https://dl.k8s.io/release/v1.23.0/bin/windows/amd64/kubectl.exe"
+        set PATH=%PATH%;$PWD
+    fi
+}
+
+function verify_minikube() {
+    minikube version
+    if [ $? -ne 0 ]; then
+        install_minikube "$1"
+    fi
+    minikube start
```

Comment:
why do we need `minikube`?

Reply:
If kubectl client is there , but its not configured , then this PR install minikube and runs kubectl against it